utils/remote_control/navigator_keyboard_control/nodes/navigator_keyboard_client.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+'''
+Keyboard Client: The keyboard client connects to the keyboard server in order
+to control NaviGator using a remote keyboard. It is able to lock control of the
+keyboard service to itself by obtaining a UUID that identifies it's service
+calls. Curses is used to display a basic UI in the terminal that gives the user
+useful feedback and captures key presses to be sent to the server.
+'''
+
+
+from __future__ import division
+
+import curses
+
+from navigator_msgs.srv import KeyboardControl
+import rospy
+
+
+__author__ = "Anthony Olive"
+__maintainer__ = "Anthony Olive"
+__email__ = "anthony@iris-systems.net"
+__copyright__ = "Copyright 2016, MIL"
+__license__ = "MIT"
+
+
+rospy.init_node('keyboard_client', anonymous=True)
+
+
+class KeyboardClient():
+
+    def __init__(self, stdscr):
+        self.screen = stdscr
+        self.num_lines = 10
+        self.screen.nodelay(True)
+        curses.curs_set(0)
+
+        self.uuid = ''
+        self.is_locked = False
+
+        self.keyboard_server = rospy.ServiceProxy('/keyboard_control', KeyboardControl)
+
+        self.help_menu = ["Lock:                 L          ",
+                          "Quit:                 q          ",
+                          "Toggle Kill:          k          ",
+                          "Force Kill:           K          ",
+                          "Station Hold:         h          ",
+                          "Autonomous Control:   u          ",
+                          "Joystick Control:     j          ",
+                          "Keyboard Control:     b          ",
+                          "Cycle Control Device: c          ",
+                          "Shooter Load:         r          ",
+                          "Shooter Fire:         f          ",
+                          "Shooter Cancel:       t          ",
+                          "The Big H:            H          ",
+                          "Move Forward:         w          ",
+                          "Move Backward:        s          ",
+                          "Move Port:            a          ",
+                          "Move Starboard:       d          ",
+                          "Yaw Counterclockwise: arrow up   ",
+                          "Yaw Clockwise:        arrow down "
+                          ]
+
+    def read_key(self):
+        '''
+        Reads the newest key from the buffer and discards all old keys that
+        were not read.
+        '''
+        keycode = -1
+        new_keycode = self.screen.getch()
+
+        # This eliminates building a buffer of keys that takes forever to process
+        while(new_keycode != -1):
```

Comment:
This while loop will only be cycling through a very small amount of keys (at most 10 for most keyboards), and will only be run when it is called by the spinner. I agree that checking for shutdown is good convention though, so I will add it.